examples/i18n/main.rs

```diff
@@ -0,0 +1,117 @@
+use std::collections::HashMap;
+use std::env;
+use std::fs::read_to_string;
+use clap::text_provider::TextProvider;
+use rust_i18n::set_locale;
+use rust_i18n::t;
+use rust_i18n::Backend;
+use std::path::PathBuf;
+use clap::{Parser, Subcommand};
+
```

Comment:
For this example, we use the rust-i18n crate to demonstrate how we can use the TextProvider API to achieve internationalization. I chose this crate because it allows looking translations at runtime, which is probably what most people would want since it avoids packaging languages into the binary that the end user will never use.